snap/backend/src/utils/accountUtils.js

```diff
@@ -25,7 +31,79 @@ const accountUtils = {
 
 		const privateKey = new PrivateKey(key.privateKeyBytes);
 		return new SymbolFacade.KeyPair(privateKey);
+	},
+	/**
+	 * Get latest metamask account index.
+	 * @param {Accounts} accounts - The accounts object.
+	 * @param {'mainnet' | 'testnet'} networkName - The network name.
+	 * @returns {number} - The latest account index.
+	 */
+	getLatestAccountIndex(accounts, networkName) {
+		return Object.values(accounts)
+			.filter(walletAccount =>
+				WalletType.METAMASK === walletAccount.account.type
+				&& networkName === walletAccount.account.networkName)
+			.reduce((maxIndex, walletAccount) => Math.max(maxIndex, walletAccount.account.addressIndex), -1);
+	},
+	async createAccount({ state, requestParams }) {
+		try {
+			const { walletLabel } = requestParams;
+			const { network, accounts } = state;
+
+			const facade = new SymbolFacade(network.networkName);
+
+			// Get the latest account index and increment it if it exists
+			const newAddressIndex = this.getLatestAccountIndex(accounts, network.networkName) + 1 || 0;
+
+			const newKeyPair = await this.deriveKeyPair(network.networkName, newAddressIndex);
+			const accountId = uuidv4();
```

Comment:
How will the accountId be used vs addressIndex?  They both are unique per account.

Reply:
You are right, both of them are unique but serve different purposes and use cases.

`accountId` - this ID represents each account.
`addressIndex` - This index is used to derive the private key (for the metamask wallet).

However, we do support importing wallet private keys, in which case `addressIndex` will be null.